server/src/clients/google.py

```diff
@@ -93,3 +94,87 @@ def get_user_info(
             parsed_user_info[field] = user_info.get(field, "")
 
         return (None, parsed_user_info)
+
+
+@error_bounded(
+    (AppError(ErrorCode.GOOGLE_API_ERROR, "Google api error - get user contacts"), None)
+)
+def _get_api_people_connections_list(
+    token: str,
+    http: Optional[httplib2.Http] = None,
+    http2: Optional[httplib2.Http] = None,
+) -> Tuple[Optional[AppError], Optional[List[Dict[str, Any]]]]:
+    """
+    Request list of user's contacts from Google People API.
+
+    Returns a list of Google API Person objects representing all
+    of the user's contacts if they exist
+    """
+    creds = Credentials(token) if token else None
+    with build("people", "v1", credentials=creds, http=http) as people_service:
+        pagination_finished = False
+        next_page_token = ""
+        api_people_list_complete = []
+        # paginate through contacts and add to list
+        while not pagination_finished:
+            people_response = (
+                people_service.people()
+                .connections()
+                .list(
+                    resourceName="people/me",
+                    personFields="names,emailAddresses,photos",
+                    pageToken=next_page_token,
+                )
+            ).execute(http=http2)
+            api_people_list_complete += people_response["connections"]
+            if "nextPageToken" in people_response:
+                next_page_token = people_response["nextPageToken"]
+            else:
+                pagination_finished = True
+
+        return (None, api_people_list_complete)
+
+
+@error_bounded(
+    (
+        AppError(
+            ErrorCode.GOOGLE_API_ERROR, "Google api error - transform user contacts"
+        ),
+        None,
+    )
+)
+def get_google_person_list(
+    token: str,
+    http: Optional[httplib2.Http] = None,
+    http2: Optional[httplib2.Http] = None,
+) -> Tuple[Optional[AppError], Optional[List[GooglePerson]]]:
+    """
+    Transform list of Google API Person objects into GooglePerson dataclass.
+
+    Returns a list of GooglePerson dataclass objects based on
+    list of Google contacts
+    """
+    error, api_people_list = _get_api_people_connections_list(token, http, http2)
+    google_person_list = []
+
+    if error:
+        return (error, None)
+
+    if api_people_list:
```

Comment:
I suppose you could return early if `api_people_list` is empty so that the chunk of code is not indented? It's more subjective though.


```suggestion
    if api_people_list == []:
        return (None, [])
```

Reply:
I like this suggestion, will adopt